### PR TITLE
CentOS 7 is no longer supported

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/config/cluster_config.py [hunk 1]
```diff
@@ -1574,8 +1574,6 @@ def _register_validators(self, context: ValidatorContext = None):  # noqa: D102
             ArchitectureOsValidator,
             os=self.image.os,
             architecture=self.head_node.architecture,
-            custom_ami=self.image.custom_ami,
-            ami_search_filters=self.dev_settings.ami_search_filters if self.dev_settings else None,
```
Comment (dreambeyondorange) [inline cli/src/pcluster/config/cluster_config.py:1578]: Why is this removed?

File: cli/src/pcluster/validators/cluster_validators.py [hunk 2]
```diff
@@ -278,21 +278,14 @@ class ArchitectureOsValidator(Validator):
     ARM AMIs are only available for a subset of the supported OSes.
     """
 
-    def _validate(self, os: str, architecture: str, custom_ami: str, ami_search_filters):
```
Comment (dreambeyondorange) [inline cli/src/pcluster/validators/cluster_validators.py:281]: Is it because this check is only used for centos?

File: cli/tests/pcluster/cli/configure/test_pcluster_configure/test_subnet_automation_no_awsbatch_no_errors_empty_vpc/pcluster.config.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Region: eu-west-1
 Image:
-  Os: centos7
```
Comment (dreambeyondorange) [inline cli/tests/pcluster/cli/configure/test_pcluster_configure/test_subnet_automation_no_awsbatch_no_errors_empty_vpc/pcluster.config.yaml:3]: should some of this be replaced with alinux2023?

**Review by dreambeyondorange — APPROVED** [review]

File: cli/src/pcluster/config/cluster_config.py (hunk 1 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/config/cluster_config.py:1578]: Because the validator was simplified. The validator was simplified because some checks on CentOS 7 are no longer needed.

File: cli/src/pcluster/validators/cluster_validators.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/validators/cluster_validators.py:281]: Yes

File: cli/tests/pcluster/cli/configure/test_pcluster_configure/test_subnet_automation_no_awsbatch_no_errors_empty_vpc/pcluster.config.yaml (hunk 3 above)
Comment (hanwen-pcluste) [reply cli/tests/pcluster/cli/configure/test_pcluster_configure/test_subnet_automation_no_awsbatch_no_errors_empty_vpc/pcluster.config.yaml:3]: These are unit tests. I thought it would be easy to mock stuffs. In integration tests, I diversified the OSes
